cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -391,14 +393,55 @@ public TimeSeries getTimeseries(String page, int pageSize, String names, String
                 query.limit(DSL.val(pageSize + 1));
             }
 
-            logger.fine(() -> query.getSQL(ParamType.INLINED));
+            SelectConditionStep<Record3<Timestamp, Double, BigDecimal>> finalQuery = query;
+            logger.fine(() -> finalQuery.getSQL(ParamType.INLINED));
 
-            query.forEach(tsRecord -> timeseries.addValue(
-                            tsRecord.getValue(dateTimeCol),
-                            tsRecord.getValue(valueCol),
-                            tsRecord.getValue(qualityNormCol).intValue()
-                    )
-            );
+            if (includeEntryDate) {
```

Comment:
This query is doubling the time it takes to retrieve time series. Can this replace the `retrieve_ts_out_tab` calls?

Reply:
While it could replace the `retrieve_ts_out_tab` call above, doing so would require implementing trim support into the query, as that is currently handled by the `retrieve_ts_out_tab` call. I haven't quite figured out the best way to do so, so maybe we can discuss this in more detail

Reply:
if the schema needs to provide more, we can update the schema call